poetry/core/masonry/builders/wheel.py

```diff
@@ -188,7 +193,12 @@ def _write_record(self, wheel):
 
             # RECORD itself is recorded with no hash or size
             csv_writer.writerow((self.dist_info + "/RECORD", "", ""))
-            f.write(record.getvalue())
+
+            record_values = record.getvalue()
```

Comment:
You should be able to rely on the compat methods that are already present to handle this instead of introducing compat logic here.